MatrixKit/Models/Room/MXKRoomBubbleCellData.m

```diff
@@ -348,6 +351,73 @@ - (void)highlightPatternInTextMessage:(NSString*)pattern withForegroundColor:(UI
     self.attributedTextMessage = nil;
 }
 
+- (void)setShouldHideSenderInformation:(BOOL)inShouldHideSenderInformation
+{
+    shouldHideSenderInformation = inShouldHideSenderInformation;
+    
+    if (!shouldHideSenderInformation)
+    {
+        // Refresh the flair
+        [self refreshSenderFlair];
+    }
+}
+
+- (void)refreshSenderFlair
+{
+    // Reset by default any observer on publicised groups by user.
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:kMXSessionDidUpdatePublicisedGroupsForUsersNotification object:self.mxSession];
+    
+    // Check first whether the room enabled the flair for some groups
+    NSArray<NSString *> *roomRelatedGroups = roomDataSource.room.state.relatedGroups;
+    if (roomRelatedGroups.count && senderId)
+    {
+        NSArray<NSString *> *senderPublicisedGroups;
+        
+        senderPublicisedGroups = [self.mxSession publicisedGroupsForUser:senderId];
+        
+        if (senderPublicisedGroups.count)
+        {
+            // Cross the 2 arrays to keep only the common group ids
+            NSMutableArray *flair = [NSMutableArray arrayWithCapacity:roomRelatedGroups.count];
+            
+            for (NSString *groupId in roomRelatedGroups)
+            {
+                if ([senderPublicisedGroups indexOfObject:groupId] != NSNotFound)
+                {
+                    MXGroup *group = [self.mxSession groupWithGroupId:groupId];
+                    if (!group)
+                    {
+                        // The current user doesn't belong to this group.
+                        // Create a group instance.
+                        group = [[MXGroup alloc] initWithGroupId:groupId];
+                    }
+                    
+                    [flair addObject:group];
+                    
+                    // Refresh the group profile from server.
+                    [self.mxSession updateGroupProfile:group success:nil failure:^(NSError *error) {
+                        
+                        NSLog(@"[MXKRoomBubbleCellData] refreshSenderFlair: group profile update failed %@", groupId);
+                        
+                    }];
+                }
+            }
+            
+            if (flair.count)
+            {
+                self.senderFlair = flair;
+            }
+            else
+            {
+                self.senderFlair = nil;
+            }
+        }
+        
+        // Observe any change on publicised groups for the message sender
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(didMXSessionUpdatePublicisedGroupsForUsers:) name:kMXSessionDidUpdatePublicisedGroupsForUsersNotification object:self.mxSession];
```

Comment:
Is there a mechanism to refresh the sender flair if the room has been added to a group while the app was in background?
I ask for this particular use case because of the group API design, we only update group data only when the app comes back to foreground.

Reply:
@manuroe: the rooms added to a group are ignored for the moment in the flair process.
Flair takes into account only the related groups defined in room settings and the publicised groups in the user's profile.